Automattic/wp-calypso
# Hosting Command Palette: Add dynamic placeholders to nested commands

Related:
* https://github.com/Automattic/wp-calypso/pull/84633
* Also specificallyhttps://github.com/Automattic/wp-calypso/pull/84633#discussion_r1408992884

## Proposed Changes

This PR adds a logic for implementing dynamic placeholders that can be set for each command separately. If there is no placeholder set, the placeholder will default to `Select a site`.

At the moment, I added the placeholder only to the `Open site dashboard` command to be `Select dashboard to open` just to implement the logic. As per the discussion with @sejas, we can adjust the wording per each command later. 

## Testing Instructions

* Navigate to Calypso Live Link
* Navigate to `/sites`
* Press `cmd + k` on macOS or `ctrl + k` on Windows to open the command palette
* Observe that the placeholder on the top says `Search for commands` and **IS NOT** changed
* Click on `Open site dashboard` command
* Observe that the placeholder on the top has been changed to `Select dashboard to open`

<img width="678" alt="Screenshot 2023-11-29 at 11 35 13 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/e5917672-6d5d-464d-8d2d-418ef9099e0a">

* Press `Esc` to go back to the top-level commands
* Confirm that placeholder is set back to `Search for commands`
* Select a command that has no custom placeholder set such as `Manage DNS`
* Confirm that the placeholder displayed is set to a default `Select a site`:

<img width="642" alt="Screenshot 2023-11-29 at 11 34 43 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/5ab4908d-ec31-4bfe-b2fd-3b89221ffb37">

## Pre-merge Checklist
